hugolib/site.go

```diff
@@ -1431,27 +1432,47 @@ func (s *Site) assembleMenus() {
 	}
 
 	sectionPagesMenu := s.Info.sectionPagesMenu
-	sectionPagesMenus := make(map[string]interface{})
-	//creating flat hash
 	pages := s.Pages
-	for _, p := range pages {
 
-		if sectionPagesMenu != "" {
+	if sectionPagesMenu != "" {
+		// Create menu entries for section pages with content
+		for _, p := range pages {
+			if p.Section() != "" && p.Kind == KindSection {
+				me := MenuEntry{Identifier: p.Section(),
+					Name:   p.LinkTitle(),
+					Weight: p.Weight,
+					URL:    p.RelPermalink()}
+
+				// menu with same id defined in config, let that one win
+				if _, ok := flat[twoD{sectionPagesMenu, me.KeyName()}]; ok {
+					continue
+				}
+				flat[twoD{sectionPagesMenu, me.KeyName()}] = &me
+			}
+		}
+
+		// Create entries for remaining content-less section pages
```

Comment:
Not sure what you mean by "content-less" section pages? The code below should be superfluous with the code above.